### PR TITLE
Claim Tracker: Create application

### LINKED ISSUE #3362: Claim Tracker: Implement REST endpoint to accept request to track claim
# Summary
Create a new service in VRO that accepts a request to track a claim with respect to a feature flag.

# Implementation Details
Details can be found in the design presentation [Feature Effects on Claim Lifecycles](https://docs.google.com/presentation/d/1xA5-254JhOh_4bsc4CADnQKkstllpkt8gpNyt3u9WDo/edit?pli=1#slide=id.g2f38420da69_0_301).

# Acceptance Criteria
- [x] New service is stood up in VRO
- [ ] The database table and orm to save the claim ID + meta is created
- [x] The service creates a REST endpoint that accepts request and saves model to table
- [ ] Unit tests

### PR DESCRIPTION
**Pull request by dfitchett**
## Summary

Creates the new 

Associated tickets or Slack threads:
- #3362 

## How does this fix it?[^1]

* creates the base skeleton of the claim tracker application
* creates the rest endpoint to add claims to track
* creates the database table for `claims.tracked_claims`
* sets up code quality checks using ruff, isort, mypy, and runs black formatting
* creates unit tests for health check endpoint and track claim endpoint

## How to test this PR
- download branch and cd to root dir
- run `./gradlew :domain-bie-events:svc-claim-tracker:check`
- run `./gradlew clean docker`
- run `COMPOSE_PROFILES="platform" ./gradlew :dockerComposeUp`
- run `COMPOSE_PROFILES="all" ./gradlew :domain-bie-events:dockerComposeUp`
- send curl: 
```
curl -L 'http://localhost:8150/track/v1/claim' -H 'Content-Type: application/json' -d '{
    "claim_id": 123,
    "established_at": "2024-01-01T00:00:00Z",
    "feature_name":"feature",
    "feature_enabled":true
}'
```
[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

File: domain-bie-events/svc-claim-tracker/README.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
+# Max Claim for Increase (CFI) API
```
Comment (CorySohrakoffUSDS) [inline domain-bie-events/svc-claim-tracker/README.md]: Update title

File: domain-bie-events/svc-claim-tracker/README.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+[Platform Base + API-Gateway](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Docker-Compose#platform-base)
+then run the svc-claim-tracker with the following command from the `abd_vro directory`:
+
+```
+COMPOSE_PROFILES="all" ./gradlew :domain-bie-events:dockerComposeUp
+```
+
+### Verify API in API Gateway
+
+Navigate to [Swagger](http://localhost:8060/webjars/swagger-ui/index.html?urls.primaryName=3.%20Max%20CFI%20API)
```
Comment (CorySohrakoffUSDS) [inline domain-bie-events/svc-claim-tracker/README.md]: Update URL?

File: db-init/src/main/resources/database/migrations/V2.2__Tracked_Claim.sql [hunk 3]
```diff
@@ -0,0 +1,12 @@
+CREATE TABLE IF NOT EXISTS "tracked_claims" (
+    id uuid NOT NULL,
+	claim_id INT NOT NULL,
+	created_at TIMESTAMP WITHOUT TIME ZONE,
```
Comment (lisac) [inline db-init/src/main/resources/database/migrations/V2.2__Tracked_Claim.sql]: double-checking: is it correct that we do not want to set a constraint of `NOT NULL` on `created_at`?

File: domain-bie-events/.pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    # Ruff version.
+    rev: v0.4.4
+    hooks:
+      # Run the linter.
+      - id: ruff
+      # Run the formatter.
+      - id: ruff-format
+
+  - repo: https://github.com/zricethezav/gitleaks
+    rev: v8.15.0
```
Comment (lisac) [inline domain-bie-events/.pre-commit-config.yaml:42]: flagging: we might want a follow-up ticket around reviewing/bumping the version numbers for these repos, across `.pre-commit-config.yaml` files in the repo. for `gitleaks`, there are more recent versions than [v8.15.0](https://github.com/gitleaks/gitleaks/releases/tag/v8.15.0) (Oct 2022).

File: domain-bie-events/svc-claim-tracker/README.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+python3 src/python_src/pull_api_documentation.py
+```
+
+## Docker Stuff
+
+### Build the image
+
+Follow steps for
+[Platform Base](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Docker-Compose#platform-base)
+then run the svc-claim-tracker with the following command from the `abd_vro directory`:
```
Comment (lisac) [inline domain-bie-events/svc-claim-tracker/README.md]:
```suggestion
then run the svc-claim-tracker with the following command from the `abd_vro` directory:
```

File: domain-bie-events/svc-claim-tracker/src/app/api.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    init_db()
+    yield
+
+
+app = FastAPI(
+    title='Claim Tracker for BIE Claim Events',
+    description='This API accepts requests to track a claim lifecycle by BIE Claim Events.',
+    contact={},
+    version='v0.1',
+    license={'name': 'CCO 1.0', 'url': 'https://github.com/department-of-veterans-affairs/abd-vro/blob/master/LICENSE.md'},
```
Comment (lisac) [inline domain-bie-events/svc-claim-tracker/src/app/api.py]:
```suggestion
    license={'name': 'CCO 1.0', 'url': 'https://github.com/department-of-veterans-affairs/abd-vro/blob/main/LICENSE.md'},
```
(branch was renamed to `main`)

**Review by lisac — APPROVED** [review]
nice work, Derek!  some minor comments, none should block the merge.

**Comment by ponny-m** [comment]
@dfitchett What was the outcome of you testing these changes locally?

**Comment by dfitchett** [comment]
> @dfitchett What was the outcome of you testing these changes locally?

@Ponnia-M It runs no problems. I'll update the directions above.

**Review by ponny-m — APPROVED** [review]
Tested this branch locally and it was successful. Great work!

**Review by brostk — APPROVED** [review]

**Pull request merged**